ONSdigital/eq-questionnaire-runner
# Fix placeholder transform resolution issue in calculation summary pages

### What is the context of this PR?
Fixes an issue where placeholders were not being evaluated correctly on Calculation Summary pages (CS, GCS) as the dependency map `_placeholder_transform_section_dependencies_by_block` was not being being populated correctly for these block types. 

Previously, Calculation Summary blocks were only accounting for any placeholders set in the specific CS/GCS block in the schema, but Calculated Summary pages can render question text (containing placeholders) from their dependent questions, so the Calculation Summary page may have a dependency on another block which then may have dependencies on other sections. The logic has now been updated to account for this and update the dependency map correctly.

The code was added to support Grand Calculated Summaries but at the moment it is not possible to add any tests for this as unlike Calculated Summaries they do not render question text from their dependent blocks.

### How to review

Use the `test_placeholder_dependencies_with_calculation_summaries` schema to test the change.
1. Launch `test_placeholder_dependencies_with_calculation_summaries`
2. Don’t accept default reporting period, enter new values and complete section 2
3. Go back to Section 1 summary screen, change dates back to default values by answering ‘Yes, I can report for this period’ for the first question.
4. Go back to Section 2 and get to the calculated summary by changing one of the answers that are used on the CS.
5. Check that the dates set in the metadata are rendered on the Calculated Summary page, not the dates that were previously entered by the user.

### Checklist
